### DISCUSSION TITLE
ed-style editing?

### DISCUSSION THREAD

**Discussion by askreet** [helix-editor/helix#4143] [Ideas] [answered]
As a vim user, I can't seem to find equivalent options for global edits like `:v/some-text/d` (delete all lines not matching some-text). Has there been discussion about adding ed-style commands to the editor. Best I can tell, they wouldn't conflict with any of the modal editing goals.

**Comment by Omnikar** [comment, marked as answer]
These sorts of commands conflict with the design goal of seeing the text you act on before you act on it. For this request in particular, you can do this:
1. Select the file (`%`)
2. Split selection into lines (`A-s`)
3. Deselect selections matching regex (`A-K`) and type your pattern into the prompt
4. Delete selections (`d`)

**Reply by askreet** [reply]
Thanks for getting back to me. I think this makes sense. One thing that is tough with combo keys like this is that there's no context to guide the user these are available (unlike, say, when you press Space or M and get a nice popup). I don't have a concrete suggestion about how that might work. Perhaps I could have searched the command menu and found these actions if I'd been accustomed to the idea of starting with whole document selection, etc.
